synapse/push/pusher.py

```diff
@@ -1,10 +1,14 @@
 from httppusher import HttpPusher
 
-PUSHER_TYPES = {
-    'http': HttpPusher
-}
-
 
 def create_pusher(hs, pusherdict):
+    PUSHER_TYPES = {
+        "http": HttpPusher,
+    }
+
+    if hs.config.email_enable_notifs:
+        from emailpusher import EmailPusher
```

Comment:
But why do you want it conditional? I'd rather have the external libraries conditionally imported, that way the tests will at least import the `EmailPusher` and check that is syntactically correct.